torch/jit/__init__.py

```diff
@@ -2034,36 +2035,42 @@ def _check_no_signature(func):
         qual_name = _qualified_name(func)
         raise RuntimeError("Must explicitly add type annotations to overloaded functions: {}".format(qual_name))
 
-def _get_overload_decl_and_defaults(func):
+def _check_no_default_args(func):
```

Comment:
There are many errors we throw in python, and this is a very simple function. I don't think we need to move it all to C++. Agreed source range should be a part of the error here though, will update it to use `FrontendError` so it does.

Reply:
It's not so much about the the complexity, it would just be nice to put error checking in one place so future readers don't need to figure out whether it was already done in the other place.